examples/usercmodule/user_c_modules.mk

```diff
@@ -0,0 +1,12 @@
+MODULES_PATH := ../../examples/usercmodule
+MODULES := cexample
+MODULES += cppexample
+MODULES += subpackage
+
+$(info USER_C_MODULES found in $(MODULES_PATH): $(MODULES))
+$(foreach module, $(MODULES), \
+    $(eval USERMOD_DIR = $(patsubst %/,%,$(MODULES_PATH))/$(module))\
+    $(info Including User C Module from $(USERMOD_DIR))\
+	$(eval include $(USERMOD_DIR)/micropython.mk)\
+)
```

Comment:
For this example, I'd suggest keeping it as minimal as possible, and similar to the cmake version.  Eg just:
```
include cexample/micropython.mk
include cppexample/micropython.mk
include subpackage/micropython.mk
```